feature request: exit with non-zero code upon validation errors

**Is your feature request related to a problem? Please describe.**

Make the CLI exit with non-zero code if validation fails

Current behavior:

```
$ istanbul-badges-readme
Info: 0. Istanbul Badges Readme process started
Info: 1. Config check process started
- Readme file exists... ✔️.
Error: Coverage file does not exist in ./coverage/coverage-summary.json
Error: Please refer to the documentation
Error: https://github.com/olavoparno/istanbul-badges-readme/blob/master/README.md
Info: 0. Istanbul Badges Readme process finished
Done in 0.17s.
$ echo $?
0
```

Exiting with 0 prevent other tools (like GitHub Actions) to detect any validation errors of `istanbul-badges-readme` easily

**Describe the solution you'd like**

adding `process.exitCode = 1;` below will do the trick work but I'm pretty sure it will be a breaking change as some users could be relying on istanbul-badges-readme success exit code

https://github.com/olavoparno/istanbul-badges-readme/blob/805ab2e19931201fb80a949d1c9693ea29aac00d/src/factory.ts#L20-L24

thus, to avoid breaking changes, this feat could be optional and enabled via `--exit-code` flag, like [`git-diff` do](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code)

**Describe alternatives you've considered**

In my use case, the coverage directory might not exists and I need to report this as an error. I'm doing the following:

```bash
$ [ -d './coverage' ] && istanbul-badges-readme || exit 1
```

but I think it would be cool if `istanbul-badges-readme` abstracts this for us :)

I'm not sure if I'm using the flag correctly (or understood it correctly), but it always fails:

```
> istanbul-badges-readme --logo='jest' "--ci" "--exitCode=1"

Istanbul Badges Readme process started
Config check process started
- Readme file exists... ✔️.
- Coverage file exists... ✔️.
- Coverage hashes exist... ✔️.
- Readme hashes exist... ✔️.
Config check process ended
Editor process started
- Getting readme hashes...
- Getting new readme data...
- Getting coverage badge url for statements...
- Writing new readme data...
Editor process ended
Istanbul Badges Readme process finished
Error: Process completed with exit code 1.
```

--- a/src/factory.ts
+++ b/src/factory.ts
@@ -22,13 +22,10 @@ export const badgerFactory =
         logError(`${error}`);
         logError('Please refer to the documentation');
         logError('https://github.com/olavoparno/istanbul-badges-readme/blob/master/README.md');
+
+        process.exitCode = getExitCodeOnError?.() ?? 0;
       })
       .finally(() => {
-        const exitCode = getExitCodeOnError?.();
         logInfo('Istanbul Badges Readme process finished');
-
-        if (exitCode) {
-          process.exit(exitCode);
-        }
       });
   };

